### PR TITLE
Fix #244: Remove error from error list window when light bulb fix is applied.

### LINKED ISSUE #244: Remove error from error list when user applies a light bulb fix
@michaelcfanning @GabeDeBacker

### PR DESCRIPTION
**Pull request by lgolding**

### CODE REVIEW

File: scripts/New-VersionConstantsFile.ps1 [hunk 1]
```diff
@@ -29,6 +29,7 @@ namespace $Namespace
         public const string Version = AssemblyVersion + Prerelease;
     }
 }
```
Comment (lgolding) [inline scripts/New-VersionConstantsFile.ps1:31]: Blank line at end to conform to new `.editorconfig`.

File: src/Sarif.Viewer.VisualStudio/ErrorList/SarifTableDataSource.cs [hunk 2]
```diff
@@ -153,13 +153,13 @@ public void ClearErrorsForLogFiles(IEnumerable<string> logFiles)
 
             using (this.tableEntriesLock.EnterWriteLock())
             {
-                IEnumerable<KeyValuePair<string, List<SarifResultTableEntry>>> logFileToTabkeEntriesToRemove = this.logFileToTableEntries.
```
Comment (lgolding) [inline src/Sarif.Viewer.VisualStudio/ErrorList/SarifTableDataSource.cs:156]: >k [](start = 91, length = 1)

Fix typo in identifier.

File: src/Sarif.Sarifer/source.extension.vsixmanifest [hunk 3]
```diff
@@ -24,6 +24,6 @@ Licensed under the MIT license. See LICENSE file in the project root for full li
     <Asset Type="Microsoft.VisualStudio.VsPackage" d:Source="Project" d:ProjectName="%CurrentProject%" Path="|%CurrentProject%;PkgdefProjectOutputGroup|" />
   </Assets>
   <Prerequisites>
-    <Prerequisite Id="Microsoft.VisualStudio.Component.CoreEditor" Version="[15.0,16.0)" DisplayName="Visual Studio core editor" />
+    <Prerequisite Id="Microsoft.VisualStudio.Component.CoreEditor" Version="[15.0,17.0)" DisplayName="Visual Studio core editor" />
```
Comment (lgolding) [inline src/Sarif.Sarifer/source.extension.vsixmanifest:27]: >17 [](start = 82, length = 2)

Without this change, the VSIX won't install on a machine where VS 2017 is not installed.

File: src/Sarif.Viewer.VisualStudio/ErrorList/SarifTableDataSource.cs [hunk 4]
```diff
@@ -190,6 +190,41 @@ public void CleanAllErrors()
...
+                if (tableEntryListWithSpecifiedError != null)
+                {
+                    // Any give error object can appear at most once in any log file, even if the
+                    // log file reports the "same" error multiple times. And we've already seen
+                    // that this error object does appear in this log file, so calling Single is
+                    // just fine.
+                    SarifResultTableEntry entryToRemove = tableEntryListWithSpecifiedError.Single(tableEntry => tableEntry.Error == errorToRemove);
+
+                    this.CallSinks(
+                        sink => sink.RemoveEntries(new List<SarifResultTableEntry> { entryToRemove }));
```
Comment (GabeDeBacker) [inline src/Sarif.Viewer.VisualStudio/ErrorList/SarifTableDataSource.cs:219]: >new List<SarifResultTableEntry> { entryToRemove }) [](start = 51, length = 50)

surprised new [] { entryToRemove } isn't sufficient :)

**Review by GabeDeBacker — APPROVED** [review]
:shipit:

**Review by michaelcfanning — APPROVED** [review]
:shipit:

File: src/Sarif.Viewer.VisualStudio/ErrorList/SarifTableDataSource.cs (hunk 4 above)
Comment (lgolding) [reply src/Sarif.Viewer.VisualStudio/ErrorList/SarifTableDataSource.cs:219]: I see this has been merged, so I'll do it in the next PR.

---
In reply to: [516775691](https://github.com/microsoft/sarif-visualstudio-extension/pull/249#discussion_r516775691) [](ancestors = 516775691)

**Pull request merged**